emission/core/wrapper/suggestion_sys.py

```diff
@@ -437,23 +438,32 @@ def review_start_loc_nominatim(lat, lon):
 11: "AIR_ON_HSR"
 '''
 
-def calculate_yelp_server_suggestion_singletrip_nominatim(tripid):
-    return_obj = { 'message': "Good job walking and biking! No suggestion to show.", 'method' : 'bike'}
-    start_lat, start_lon = geojson_to_lat_lon_separated(tripid["start_loc"])
-    end_lat, end_lon = geojson_to_lat_lon_separated(tripid["end_loc"])
+
+def calculate_yelp_server_suggestion_singletrip_nominatim(tripid, uuid):
+    all_users = pd.DataFrame(list(edb.get_uuid_db().find({}, {"uuid": 1, "_id": 0})))
+    user_id = all_users.iloc[all_users[all_users.uuid == uuid].index.tolist()[0]].uuid
+    timeseries = esta.TimeSeries.get_time_series(user_id)
+    cleaned_trips = timeseries.get_data_df("analysis/cleaned_trip", time_query = None)
+    spec_trip = cleaned_trips.iloc[cleaned_trips[cleaned_trips._id == tripid].index.tolist()[0]]
```

Comment:
Again, this is the kind of code you would write in an exploratory setting (e.g. in `ipython`), not in a script. You are reading all trips into memory and then retrieving the trips that matches. This is fine if the user has 100 trips, but what happens if the user has 10,000 trips? You should really query directly using the `trip_id` as well. Look at the functions in `abstract_timeseries` to see what else you can use.

Reply:
or I think that there might be some decorators in `emission.storage.decorations` that might be useful.

Reply:
See https://github.com/e-mission/e-mission-server/pull/620#discussion_r244630629